mongorover/MongoCollection.lua

```diff
@@ -270,15 +270,15 @@ local MongoCollection = {__mode="k"}
 	-- Example usage at @{indexing.lua}.
 	-- @tparam table of fields and their corresponding index types
 	-- @tparam table of options to be used when creating index
-	-- @return string index name on success
+	-- @return string index name
 	function MongoCollection:createIndex(keys,opt)
 	    return self.collection_t:collection_create_index(luaBSONObjects, keys, opt)
 	end
 
 	---
 	-- Delete an index for a specified collection.
 	-- Example usage at @{indexing.lua}.
-	-- @tparam string an index name to be deleted
+	-- @tparam string an index name or original table with indexed fields and their corresponding types
```

Comment:
Sorry for being unclear. If it is not successful, an error should be thrown. If it is successful, just let it continue on and return nil.

```
>>> client.foo.bar.drop_index('this_will_fail')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/christopherjwang/.virtualenvs/junk/lib/python2.7/site-packages/pymongo/collection.py", line 1631, in drop_index
    parse_write_concern_error=True)
  File "/Users/christopherjwang/.virtualenvs/junk/lib/python2.7/site-packages/pymongo/collection.py", line 232, in _command
    collation=collation)
  File "/Users/christopherjwang/.virtualenvs/junk/lib/python2.7/site-packages/pymongo/pool.py", line 419, in command
    collation=collation)
  File "/Users/christopherjwang/.virtualenvs/junk/lib/python2.7/site-packages/pymongo/network.py", line 116, in command
    parse_write_concern_error=parse_write_concern_error)
  File "/Users/christopherjwang/.virtualenvs/junk/lib/python2.7/site-packages/pymongo/helpers.py", line 210, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: index not found with name [this_will_fail]
```

The value from `mongoc_collection_keys_to_index_string()` is only valuable because that will tell us whether there was an error or not. C doesn't have exceptions.